Code/Framework/AzFramework/AzFramework/Archive/Archive.cpp

```diff
@@ -1241,6 +1241,7 @@ namespace AZ::IO
 
             m_arrZips.insert(revItZip.base(), desc);
 
+            lock.unlock();
```

Comment:
Can you add a brief comment on why this is needed?

Reply:
Good call, I've added it